Set up 2x Macmini9,1 devices as Monterey JSC EWS queues

chgibb-apple opened the pull request:
#### 2b352bffd07c27ab0c557a17b746155579a06454
<pre>
Set up 2x Macmini9,1 devices as Monterey JSC EWS queues
<a href="https://bugs.webkit.org/show_bug.cgi?id=248991">https://bugs.webkit.org/show_bug.cgi?id=248991</a>
rdar://102326975

Reviewed by Ryan Haddad.

Set up 2x Macmini9,1 devices as Monterey JSC EWS queues

* Tools/CISupport/ews-build/config.json:
* Tools/CISupport/ews-build/factories_unittest.py:
(TestExpectedBuildSteps):
* Tools/CISupport/ews-build/steps_unittest.py:
(TestCheckChangeRelevance):
(TestCheckChangeRelevance.test_relevant_jsc_arm64_patch):

Canonical link: <a href="https://commits.webkit.org/257656@main">https://commits.webkit.org/257656@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/a20b56003052706bd059eeec9d5c10bbae73df36

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/99612 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/8787 "Built successfully") | [  ~~🛠 mac~~](https://ews-build.webkit.org/#/builders/43/builds/32700 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/108971 "Built successfully") | [  ~~🛠 🧪 win~~](https://ews-build.webkit.org/#/builders/10/builds/169206 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/103608 "Passed tests") | [  ~~🛠 ios-sim~~](https://ews-build.webkit.org/#/builders/76/builds/9369 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 mac-AS-debug~~](https://ews-build.webkit.org/#/builders/61/builds/86084 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 gtk~~](https://ews-build.webkit.org/#/builders/36/builds/92082 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/12/builds/106891 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [  ~~🧪 webkitperl~~](https://ews-build.webkit.org/#/builders/19/builds/105386 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/76/builds/9369 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/43/builds/32700 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/36/builds/92082 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/76/builds/9369 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/43/builds/32700 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/36/builds/92082 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/81/builds/2618 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/43/builds/32700 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| [✅ 🧪 services](https://ews-build.webkit.org/#/builders/20/builds/98543 "Passed tests") | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/2557 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/61/builds/86084 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/5268 "Built successfully and passed tests") | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/80/builds/8697 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/43/builds/32700 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/4420 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

ryanhaddad commented on Tools/CISupport/ews-app/ews/views/statusbubble.py:
```diff
@@ -45,7 +45,7 @@ class StatusBubble(View):
     ALL_QUEUES = ['style', 'ios', 'ios-sim', 'mac', 'mac-AS-debug', 'tv', 'tv-sim', 'watch', 'watch-sim', 'gtk', 'wpe', 'wincairo', 'win',
                   'ios-wk2', 'mac-wk1', 'mac-wk2', 'mac-wk2-stress', 'mac-AS-debug-wk2', 'gtk-wk2', 'api-ios', 'api-mac', 'api-gtk',
-                  'bindings', 'jsc', 'jsc-armv7', 'jsc-armv7-tests', 'jsc-mips', 'jsc-mips-tests', 'jsc-i386', 'webkitperl', 'webkitpy', 'services']
+                  'bindings', 'jsc', 'JSC-Tests-arm64-EWS', 'jsc-armv7', 'jsc-armv7-tests', 'jsc-mips', 'jsc-mips-tests', 'jsc-i386', 'webkitperl', 'webkitpy', 'services']
```
```suggestion
                  'bindings', 'jsc', 'jsc-arm64', 'jsc-armv7', 'jsc-armv7-tests', 'jsc-mips', 'jsc-mips-tests', 'jsc-i386', 'webkitperl', 'webkitpy', 'services']
```

ryanhaddad commented on Tools/CISupport/ews-app/ews/common/github.py:
```diff
@@ -204,7 +204,7 @@ class GitHubEWS(GitHub):
                           ['webkitperl', 'ios-wk2', 'mac-AS-debug', 'gtk-wk2', ''],
                           ['webkitpy', 'api-ios', 'api-mac', 'api-gtk', ''],
-                          ['jsc', 'tv', 'mac-wk1', 'jsc-armv7', ''],
+                          ['jsc', 'tv', 'mac-wk1', 'jsc-armv7', 'JSC-Tests-arm64-EWS', ''],
```
I think this should be named `jsc-arm64`, but also I think we might need to make a new list because each line represents a row, and each item in the list represents a column in the table @aj062

ryanhaddad commented on Tools/CISupport/ews-build/config.json:
```diff
@@ -246,6 +248,12 @@
       "workernames": ["ews127", "ews128"]
     },
+    {
+      "name": "JSC-Tests-arm64-EWS", "shortname": "jsc", "icon": "buildAndTest",
```
```suggestion
      "name": "JSC-Tests-arm64-EWS", "shortname": "jsc-arm64", "icon": "buildAndTest",
```

chgibb-apple replied:
You've recommended renaming from `JSC-Tests-arm64-EWS` to `jsc-arm64` in a subset of all of the references to `JSC-Tests-arm64-EWS`. Is this by design?

ryanhaddad replied:
Yes, for consistency with the short names for all the other queues which are lower case and very terse.

chgibb-apple replied:
Sounds good. I think I've made the relevant changes.

chgibb-apple committed:
Set up 2x Macmini9,1 devices as Monterey JSC EWS queues
https://bugs.webkit.org/show_bug.cgi?id=248991
rdar://102326975

Reviewed by Ryan Haddad.

Set up 2x Macmini9,1 devices as Monterey JSC EWS queues

* Tools/CISupport/ews-build/config.json:
* Tools/CISupport/ews-build/factories_unittest.py:
(TestExpectedBuildSteps):
* Tools/CISupport/ews-build/steps_unittest.py:
(TestCheckChangeRelevance):
(TestCheckChangeRelevance.test_relevant_jsc_arm64_patch):

Canonical link: https://commits.webkit.org/257656@main
